src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java

```diff
@@ -122,5 +132,25 @@ public void run() {
                 }
             }
         }
+
+        private void checkPermission(Permission permission) {
+            GitLabConnectionConfig gitlabConfig =
+                    (GitLabConnectionConfig) Jenkins.get().getDescriptor(GitLabConnectionConfig.class);
+            if (gitlabConfig != null) {
+                if (gitlabConfig.isUseAuthenticatedEndpoint()) {
+                    if (!project.getACL().hasPermission(authentication, permission)) {
```

Comment:
'hasPermission(org.acegisecurity.Authentication, hudson.security.Permission)' is deprecated, suggest using https://javadoc.jenkins-ci.org/hudson/security/AccessControlled.html#hasPermission2(org.springframework.security.core.Authentication,hudson.security.Permission) instead.